app/routes/session.py

```diff
@@ -50,39 +54,53 @@ def login():
 
     validate_jti(decrypted_token)
 
-    try:
-        runner_claims = validate_runner_claims(decrypted_token)
-    except ValidationError as e:
-        raise InvalidTokenException("Invalid runner claims") from e
+    runner_claims = get_runner_claims(decrypted_token)
+
     # pylint: disable=assigning-non-slot
     g.schema = load_schema_from_metadata(metadata=runner_claims)
     schema_metadata = g.schema.json["metadata"]
 
-    try:
-        questionnaire_claims = validate_questionnaire_claims(
-            decrypted_token, schema_metadata
-        )
-    except ValidationError as e:
-        raise InvalidTokenException("Invalid questionnaire claims") from e
+    questionnaire_claims = get_questionnaire_claims(
+        decrypted_token=decrypted_token, schema_metadata=schema_metadata
+    )
+
+    theme = g.schema.json["theme"]
+    ru_ref = None
+    questionnaire_id = None
 
-    claims = {**runner_claims, **questionnaire_claims}
+    if decrypted_token.get("version"):
+
+        for key, value in questionnaire_claims.items():
+            runner_claims["survey_metadata"]["data"][key] = value
+        claims = runner_claims
+
+        if theme == "social":
+            questionnaire_id = (
+                claims.get("survey_metadata").get("data").get("questionnaire_id")
+            )
+        else:
+            ru_ref = claims.get("survey_metadata").get("data").get("ru_ref")
```

Comment:
Theme doesn't matter, this could just be:
```py
data = claims.get("survey_metadata").get("data", {})
ru_ref = data.get("ru_ref")
questionnaire_id = data.get("questionnaire_id")
```